corehq/apps/app_manager/management/commands/migrate_case_list_custom_variables.py

```diff
@@ -0,0 +1,74 @@
+from corehq.apps.app_manager.management.commands.helpers import AppMigrationCommandBase
+from corehq.toggles import CASE_LIST_CUSTOM_VARIABLES
+
+from lxml import etree
+
+
+class Command(AppMigrationCommandBase):
+    help = """Migrate case list custom variables from xml to dict and back"""
+
+    include_builds = True
+    include_linked_apps = True
+    DOMAIN_LIST_FILENAME = 'migrate_case_list_custom_variables_domain.txt'
+    DOMAIN_PROGRESS_NUMBER_FILENAME = 'migrate_case_list_custom_variables_progress.txt'
+    chunk_size = 1
+
+    def add_arguments(self, parser):
+        super(Command, self).add_arguments(parser)
+
+        parser.add_argument(
+            '--reverse',
+            action='store_true',
+            default=False,
+            help="Perform the migration in reverse",
+        )
+
+    def get_domains(self):
+        return CASE_LIST_CUSTOM_VARIABLES.get_enabled_domains()
+
+    @staticmethod
+    def parse(xml):
+        elems = etree.fromstring(f"<w>{xml}</w>")
+        return {e.tag: e.get("function") for e in elems}
+
+    @staticmethod
+    def migrate_detail_forward(detail):
+        if detail and (detail.get('custom_variables') is not None):
+            detail['custom_variables_dict'] = Command.parse(detail['custom_variables'])
+            del detail['custom_variables']
```

Comment:
@esoergel If I understand [this guide](https://commcare-hq.readthedocs.io/migrations_in_practice.html) correctly I should remove the `del detail['custom_variables']` and do it in a separate migration later?